#!/bin/bash
# verifybamid_v2.0.1 2.0.1
# Generated by dx-app-wizard.
#
# Basic execution pattern: Your app will run on a single machine from
# beginning to end.
#
# Your job's input variables (if any) will be loaded as environment
# variables before this script runs.  Any array inputs will be loaded
# as bash arrays.
#
# Any code outside of main() (or any entry point you may add) is
# ALWAYS executed, followed by running the entry point itself.
#
# See https://documentation.dnanexus.com/developer for tutorials on how
# to modify this file.

# The following line causes bash to exit at any point if there is any error
# and to output each line as it is executed -- useful for debugging
set -e -x -o pipefail

# Store the bam file name as a string
bam_file=`dx describe "${input_bam}" --name`

# Remove .bam extension from bam file name
bam_prefix="${bam_file%.bam}"

# Download bam and index files
dx download "$input_bam" -o "$bam_file"
dx download "$input_bam_index" -o "${bam_prefix}.bai" 
dx download "$vcf_file" -o vcf_file

# Create output directory
mkdir -p out/verifybamid_out/QC/

# Call verifyBamID for contamination check. The following notable options are passed:
# --ignoreRG; to check the contamination for the entire BAM rather than examining individual read groups
# --precise; calculate the likelihood in log-scale for high-depth data (recommended when --maxDepth is greater than 20)
# --maxDepth 1000; For the targeted exome sequencing, --maxDepth 1000 and --precise is recommended.
verifyBamID --vcf vcf_file \
     --bam $bam_file \
     --out out/verifybamid_out/QC/$bam_prefix \
     --verbose --ignoreRG --precise --maxDepth 1000

# Upload results to DNAnexus
dx-upload-all-outputs